client-spark/common/src/main/java/org/apache/spark/shuffle/RssSparkShuffleUtils.java

```diff
@@ -149,16 +156,31 @@ public static int estimateTaskConcurrency(SparkConf sparkConf) {
     if (!sparkConf.getBoolean(Constants.SPARK_DYNAMIC_ENABLED, false)) {
       int executorInstances = sparkConf.getInt(Constants.SPARK_EXECUTOR_INSTANTS,
           Constants.SPARK_EXECUTOR_INSTANTS_DEFAULT_VALUE);
-      taskConcurrency =  executorInstances > 0 ? executorInstances * taskConcurrencyPerExecutor : 0;
+      taskConcurrency = executorInstances > 0 ? executorInstances * taskConcurrencyPerExecutor : 0;
     } else {
       // Default is infinity
       int maxExecutors = Math.min(sparkConf.getInt(Constants.SPARK_MAX_DYNAMIC_EXECUTOR,
           Constants.SPARK_DYNAMIC_EXECUTOR_DEFAULT_VALUE), Constants.SPARK_MAX_DYNAMIC_EXECUTOR_LIMIT);
       int minExecutors = sparkConf.getInt(Constants.SPARK_MIN_DYNAMIC_EXECUTOR,
           Constants.SPARK_DYNAMIC_EXECUTOR_DEFAULT_VALUE);
-      taskConcurrency = (int)((maxExecutors - minExecutors) * dynamicAllocationFactor + minExecutors)
-                            * taskConcurrencyPerExecutor;
+      taskConcurrency = (int) ((maxExecutors - minExecutors) * dynamicAllocationFactor + minExecutors)
+          * taskConcurrencyPerExecutor;
     }
     return taskConcurrency;
   }
+
+  public static StatefulUpgradeClientOptions getStatefulUpgradeOptions(RssConf clientConfig) {
```

Comment:
Will this config option be dispatched by the coordinator when we need to upgrade the service?

Reply:
Yes. And I think this can leverage the dynamic config mechanism